src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
+            break;
         }
+      } else {
+        filePath = join(listName, stateToCustListEnding[metaDataState]);
       }
-    }
-  };
 
-  const clearTemporaryItemsOnBack: typeof onBack = (e) => {
-    setCanWrite(false);
-    setChosenKey('C');
-    setChosenOctave(3);
-    setChosenName('');
-    rawSetChosenBuiltInList('');
-    rawSetChosenCustomListPath('');
-    setSelectedRecordingSetIndex(-1);
-    setProjectFile(DUMMY_PROJECT_FILE);
-    setShowingDetails(false);
-    setImmediate(() => onBack(e));
+      return filePath;
+    };
+
+    let isSubscribed = true;
+
+    const otoFilePath = getFilePath('oto-ini', listName, isBuiltIn);
+    const listFilePath = getFilePath('list-preview', listName, isBuiltIn);
+    const dvcfgFilePath = getFilePath('dvcfg', listName, isBuiltIn);
+
+    const listPromise = checkFileExistence(listFilePath);
+    const otoPromise = checkFileExistence(otoFilePath);
+    const dvcfgPromise = checkFileExistence(dvcfgFilePath);
+
+    Promise.all([
+      otoPromise.catch((error) => {
+        return error;
+      }),
+      listPromise.catch((error) => {
+        return error;
+      }),
+      dvcfgPromise.catch((error) => {
+        return error;
+      }),
+    ])
+      .then((responses) => {
+        const states: MetadataState[] = [];
+        if (isSubscribed) {
+          if (responses[0] == 'file') {
+            states.push('list-preview');
+            setFilePathRec(listFilePath);
+          } else {
+            setFilePathRec('');
+          }
+          if (responses[1] == 'file') {
+            states.push('oto-ini');
+            setFilePathOto(otoFilePath);
+          } else {
+            setFilePathOto('');
+          }
+          if (responses[2] == 'file') {
+            states.push('dvcfg');
+            setFilePathDvcfg(dvcfgFilePath);
+          } else {
+            setFilePathDvcfg('');
+          }
+          if (states.length == 0) {
+            setMetadataStateIndex(-1);
+          } else {
+            setDropdownStateArray(states);
+            setMetadataStateIndex(0);
+          }
+        }
+      })
+      .catch((error) => {
+        log.error(error);
+        throw error;
+      });
+    return () => {
+      isSubscribed = false;
+    };
   };
 
-  useEffect(() => {
-    onSetSelected(recordingSets[selectedRecordingSetIndex]);
-    // We have to ignore onSetSelected because it's new every time
-    // eslint-disable-next-line
-  }, [recordingSets, selectedRecordingSetIndex]);
-
-  useEffect(() => {
-    if (canWrite && recordingProject) {
-      const currentProjectState: Partial<ProjectFile> = {
-        name: recordingProject.name,
-        recordingSets,
-      };
-      writeFile(
-        join(recordingProject.rootPath, PROJECT_CONFIG_FILENAME),
-        JSON.stringify(currentProjectState, null, 2),
-      ).catch(log.error);
+  switch (recordingSetState) {
+    case 'home':
+      return (
+        <ConfigureRecordingSet
```

Comment:
This is the same as `case 'external':` it seems? You can collapse them. Or, if they are not the same, still consider collapsing them and make what's different clear.